### DISCUSSION TITLE
lack sourcemap support for local js bundles that are nested in apps(apk,ipa...)

### DISCUSSION THREAD

**Discussion by zrj570543941** [getsentry/support#78] [Support] [answered]
### Problem Statement

our company has such a project, it bundles the offline js css html assets in the apps.
those html in those app are opened with file protocal. so they have no common urlPrefix in different machines(especially between ios and android).
how should i get the sourcemap support at such a special case?
I have tried this solution, but  not work :

mock a sourcemap prefix,and upload them to the sentry server
.min.js end with a line:
//# sourceMappingURL=https://assets.dxycdn.com/gitrepo/idxyer_hybrid/dist/dist.bundle/js/answer.js.map
and upload to the sentry server 
 
![企业微信截图_97caf840-e674-4304-b321-674a2a9292ad](https://user-images.githubusercontent.com/12026663/187407003-8e8cf37c-544a-443a-a3aa-d7c506383ee2.png)

final resule is like this:
![企业微信截图_7f5fb35d-8465-40f9-8aae-ef68e585233a](https://user-images.githubusercontent.com/12026663/187408357-eebb10d9-389a-46c9-bce2-a458c1f48cb6.png)

**Comment by souredoutlook** [comment]
Hi @zrj570543941 thanks for writing in! Can you share a link to the issue in your sentry project so I can investigate this further?

It looks like you'll need to use the [ReWrite Frames integration](https://docs.sentry.io/platforms/javascript/configuration/integrations/plugin/#rewriteframes) to iterate over these frames and modify the filename to reflect the artifact names for source maps and source files you're uploading.

**Reply by zrj570543941** [reply]
my company self host sentry server. but thanks to your point, this issue is resolved. but i found a new issue. rewrite frame has no effect for rewrite file protocal js files. i have to rewrite by myself like this in beforesend event:  
```
// not work for file:///xxxx
new RewriteFramesIntegration({
    prefix: '/',
}),
// this works, rewrite by myself
beforeSend(event, hint) {
        const stacktrace = event.exception.values[0].stacktrace;
        if (stacktrace && stacktrace.frames) {
          stacktrace.frames.forEach(function (frame) {
            const filename = baseName(frame.filename);
            frame.filename = `/${filename}.js`;
          });
        }
        return event;
      }
}

```

**Comment by souredoutlook** [comment, marked as answer]
@zrj570543941 thanks for closing the loop on that. I'll re-iterate your solution below and mark as the answer for anyone that runs into this in the future:

>rewrite frame has no effect for rewrite file protocal js files. i have to rewrite by myself like this in beforesend event:

```
// not work for file:///xxxx
new RewriteFramesIntegration({
    prefix: '/',
}),
// this works, rewrite by myself
beforeSend(event, hint) {
        const stacktrace = event.exception.values[0].stacktrace;
        if (stacktrace && stacktrace.frames) {
          stacktrace.frames.forEach(function (frame) {
            const filename = baseName(frame.filename);
            frame.filename = `/${filename}.js`;
          });
        }
        return event;
      }
}
```
